scripts/generate-app-config/generateAppConfig.js

```diff
@@ -61,49 +66,64 @@ const generateAppConfig = (siteConfig, production, verbose) => {
     imports,
   );
 
-  const routingConfig = addConfig(
-    generateRouteConfig(siteConfig, generatePagesConfig(siteConfig, production, verbose)),
-    'routeConfig.js',
+  const contentConfig = generateContentConfig(siteConfig, generatePagesConfig(siteConfig, production, verbose));
+  const menuConfigImport = addConfig(
+    contentConfig.menuItems,
+    'menuItems.js',
+    buildPath,
+    fse,
+    imports,
+  );
+  const contentConfigImport = addConfig(
+    contentConfig.content,
+    'contentConfig.js',
     buildPath,
     fse,
     imports,
   );
 
-  const navigationItems = addConfig(
-    generateNavigationItems(navConfig),
+  const { navigationItems, capabilities } = generateNavigationItems(navConfig);
+  const navigationItemsImport = addConfig(
+    navigationItems,
     'navigationItems.js',
     buildPath,
     fse,
     imports,
   );
 
-  const extensions = addConfig(
-    generateExtensions(appConfig),
-    'extensions.jsx',
+  const extensionConfigImport = addConfig(
+    generateExtensionConfig(appConfig.extensions),
+    'extensionsConfig.js',
     buildPath,
     fse,
     imports,
   );
 
-  // Add any side-effect theme imports.
-  importSideEffects(themeImports, imports);
+  // Create search items file
+  writeConfig(
+    generateSearchItems(contentConfig.content),
+    'searchItems.js',
+    buildPath,
+    fse,
+  );
 
   // Add any side-effect imports.
   importSideEffects(sideEffectImports, imports);
 
   // Building out the overall config import.
   const config = {
     nameConfig,
-    utilityConfig,
-    routingConfig,
-    navigationItems,
-    extensions,
+    settingsConfig,
+    menuItems: menuConfigImport,
+    contentConfig: contentConfigImport,
+    navigationItems: navigationItemsImport,
     indexPath: navConfig.navigation.index,
-    defaultTheme: appConfig.defaultTheme,
-    themes: appConfig.themes,
+    capabilities: capabilities.config,
```

Comment:
Could we return capabilities as a flat object instead of as capabilities.config?